### PR TITLE
Add Messages from EventFlowMsg and DemoMsg

### PR DESCRIPTION
**Pull request by savage13**
Add Messages from EventFlowMsg and DemoMsg

- Messages are converted from yaml to json using `gen_messages.py`
- Only necessary messages files are fetched

The `load()` function callback was changed to `async` to get messages before the graph is built

### CODE REVIEW

File: gen_messages.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import sys
+
+if len(sys.argv) < 2:
+    print("Usage: gen_messages.py path-to-messages")
+    sys.exit(1)
+
+datafiles = sys.argv[1]
+
+
+def u_constrt(loader, node):
```
Comment (leoetlino) [inline gen_messages.py]: These don't seem to be useful

File: gen_messages.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+yaml.add_constructor('!list', list_constrt)
+yaml.add_constructor('!obj', obj_constrt)
+yaml.add_constructor('!str64', str_constrt)
+yaml.add_constructor('!str32', str_constrt)
+yaml.add_constructor('!str256', str_constrt)
+yaml.add_constructor('!vec3', vec3_constrt)
+
+dirs = ['EventFlowMsg', 'DemoMsg']
+
+for dir in ['EventFlowMsg', 'DemoMsg']:
```
Comment (leoetlino) [inline gen_messages.py]:
```suggestion
for dir in dirs:
```

File: gen_messages.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    if not os.path.exists(f"msg/{dir}"):
```
Comment (leoetlino) [inline gen_messages.py]: You can use Path (import from pathlib) to simplify this.

```py
(Path("msg") / dir).mkdir(parents=True, exist_ok=True)
```

File: gen_messages.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        os.makedirs(f"msg/{dir}", mode=0o755)
+    for file in glob.glob(f'{datafiles}/Message/Msg_USen.product.sarc/{dir}/*.msyt'):
```
Comment (leoetlino) [inline gen_messages.py]:
```suggestion
    for file in (Path(datafiles) / "Message" / "Msg_USen.product.sarc" / dir).glob("*.msyt"):
```

File: gen_messages.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        print(file)
+        data = {}
+        with open(file,'r') as f:
```
Comment (leoetlino) [inline gen_messages.py]:
```suggestion
        with path.open("r") as f:
```

File: gen_messages.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+            bdata = yaml.load(f, Loader=yaml.FullLoader)
+            base = file.replace(f'{datafiles}/Message/Msg_USen.product.sarc/','').replace('.msyt','')
+            for key, entry in bdata['entries'].items():
```
Comment (leoetlino) [inline gen_messages.py]:
```suggestion
            for key, entry in bdata["entries"].items():
```

File: gen_messages.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+for dir in ['EventFlowMsg', 'DemoMsg']:
+    if not os.path.exists(f"msg/{dir}"):
+        os.makedirs(f"msg/{dir}", mode=0o755)
+    for file in glob.glob(f'{datafiles}/Message/Msg_USen.product.sarc/{dir}/*.msyt'):
+        print(file)
+        data = {}
+        with open(file,'r') as f:
+            bdata = yaml.load(f, Loader=yaml.FullLoader)
+            base = file.replace(f'{datafiles}/Message/Msg_USen.product.sarc/','').replace('.msyt','')
```
Comment (leoetlino) [inline gen_messages.py]:
```suggestion
            base = path.stem
```

File: viewer.js [hunk 8]
```diff
@@ -16,6 +16,8 @@ const params = getParams(window.location.search);
 let graph;
 let eventNamesVisible = params.names;
 let eventParamVisible = params.params;
+let ev_messages = {};
```
Comment (leoetlino) [inline viewer.js]: evMessages

File: viewer.js [hunk 9]
```diff
@@ -64,7 +66,15 @@ function getNodeLabel(node) {
         continue;
       }
       const valueStr = typeof value === 'number' ? value.toFixed(6).replace(/\.?0*$/, '') : value;
-      label += `\n${key}: ${valueStr}`;
+        label += `\n${key}: ${valueStr}`;
+        if(key == "MessageId") {
```
Comment (leoetlino) [inline viewer.js]:
```suggestion
        if (key === 'MessageId') {
```
(please use single quotes so the coding style is more consistent)

File: viewer.js [hunk 10]
```diff
@@ -64,7 +66,15 @@ function getNodeLabel(node) {
...
+            const dash = "\n" + "-".repeat(40) + "\n";
+            const msg = ev_messages[valueStr];
+            if(msg) {
+                let text = msg.contents.filter(item => item.text).map(item => item.text).join("");
```
Comment (leoetlino) [inline viewer.js]:
```suggestion
                const text = msg.contents.map(item => item.text).filter(item => item).join('');
```

File: viewer.js [hunk 11]
```diff
@@ -333,11 +343,35 @@ function select(id) {
...
+    // Fetch all message files and add into 
+    await Promise.all(files.map(file => fetch(`msg/${file}.json`).then(res => res.json())))
+        .then(values =>
+            values.forEach(value => { ev_messages = Object.assign(ev_messages, value); })
+        );
+}
+
 function load(cb) {
-  fetch(params.data).then((res) => res.json()).then((data) => {
+
```
Comment (leoetlino) [inline viewer.js]: formatting

File: viewer.js [hunk 12]
```diff
@@ -333,11 +343,35 @@ function select(id) {
   }
 }
 
+function unique(value, index, self) {
+  return self.indexOf(value) === index;
+}
+
+async function get_messages(data) {
```
Comment (leoetlino) [inline viewer.js]: getMessages

File: viewer.js [hunk 13]
```diff
@@ -16,6 +16,8 @@ const params = getParams(window.location.search);
 let graph;
 let eventNamesVisible = params.names;
 let eventParamVisible = params.params;
+let ev_messages = {};
+
 
 const cboxShowParams = document.querySelector('#cbox-showparams');
 cboxShowParams.checked = eventParamVisible;
```
Comment (leoetlino) [inline viewer.js:24]: consider adding a checkbox for this new feature ("Show message text", maybe?)

File: gen_messages.py (hunk 1 above)
Comment (savage13) [reply gen_messages.py]: Removed.

File: gen_messages.py (hunk 2 above)
Comment (savage13) [reply gen_messages.py]: Ok

File: gen_messages.py (hunk 3 above)
Comment (savage13) [reply gen_messages.py]: Ok

File: gen_messages.py (hunk 5 above)
Comment (savage13) [reply gen_messages.py]: Ok

File: gen_messages.py (hunk 6 above)
Comment (savage13) [reply gen_messages.py]: Ok

File: gen_messages.py (hunk 7 above)
Comment (savage13) [reply gen_messages.py]: Ok. Added in the parent name to include the directory name as well

File: viewer.js (hunk 8 above)
Comment (savage13) [reply viewer.js]: Ok

File: viewer.js (hunk 9 above)
Comment (savage13) [reply viewer.js]: Ok

File: viewer.js (hunk 10 above)
Comment (savage13) [reply viewer.js]: Ok

File: viewer.js (hunk 11 above)
Comment (savage13) [reply viewer.js]: Ok

File: viewer.js (hunk 12 above)
Comment (savage13) [reply viewer.js]: Ok

File: viewer.js (hunk 13 above)
Comment (savage13) [reply viewer.js:24]: Done

**Comment by savage13** [comment]
Was a bit confused that this did not have "Reviewable" :|

**Comment by savage13** [comment]
`lib/node_modules` is missing from the repo. Not sure if that was on purpose.

**Review by leoetlino — APPROVED** [review]

**Pull request merged**